### ISSUE TITLE
Failure to summarise haplotype information

### ISSUE THREAD

**Issue by PabloOfEpidemiology** [TheSatoLab/CoVFit#2]
The script fails to complete the "Summarise haplotype information" (lines 122-125). The output below is where the console stalls and never proceeds. Most I tried was 3TB RAM for 48h with various number of CPUs (1, 4, 8).

[1] "Mutations with counts > 10000:"
# A tibble: 209 × 2
   mut     count.mut
   <chr>       <int>
 1 S:D614G  10153516
 2 S:T478K   8148431
 3 S:G142D   6405388
 4 S:N501Y   6263039
 5 S:P681H   6199130
 6 S:H655Y   5328528
 7 S:S477N   5268015
 8 S:N969K   5228084
 9 S:N679K   5227545
10 S:Q954H   5226402
# ℹ 199 more rows
[1] "Refined mutation information. Unique mutations count:"
[1] 209

To clarify, I never get to the next print statement, i.e. Haplotype summary. First few rows:

I think parallelising this step will significantly speed up the process, but wanted to ask your opinion and what were hardware settings that you used.

**Comment by AdamInTokyo** [comment]
Hi Pablo. I'm sorry to hear you're having trouble running the code in the repository. What exactly is the script file you are trying to run? None of the scripts require that much RAM, so it's not a problem of your hardware resources.

**Comment by PabloOfEpidemiology** [comment]
Hi, sorry, that I didn't clarify the script I was running. It's make_input_all_region.3.R. I'm pretty sure that my input files are correctly preprocessed, which is supported by the fact that the script proceeds as far as the output indicated above (but not further!)

**Comment by AdamInTokyo** [comment]
Hmmm. I'm surprised you're having that output from that script. I would recommend confirming that you have the most up-to-date version of the script from the repository, and set line 27 setwd() to your path. Also note that in the script, the file names for full.nextclade.tsv and full.nextclade.mut_long.tsv are prepended with `20231101.` as that was the information date at the time. Please adjust the strings with your file names. Give it one more try and let me know what the output is. Thanks.

**Comment by PabloOfEpidemiology** [comment]
Hi, sure, I'll do another run with the new script version. I was using the script from mid-April. The naming nomenclature is not an issue, I accounted for it.

I just checked the new script, and realised that the annotations that you see in the output above is because I asked my AI assistant script to annotate the code for debugging, but yeah, then the issue is in lines 76-78:
mut.info.sum <- mut.info %>% group_by(seqName) %>% summarize(haplotype = toString(mut))
mut.info.sum <- mut.info.sum %>% mutate(hap_Id = paste("hap_",as.character(as.numeric(as.factor(haplotype))),sep=""))
mut.info.sum <- mut.info.sum %>% inner_join(metadata.merged %>% dplyr::select(seqName,country),by="seqName")

I'll check if the new script version resolves the issue and will let you know. Thank you!
